lib/usd/ui/layerEditor/stageSelectorWidget.cpp

```diff
@@ -112,25 +112,40 @@ void StageSelectorWidget::selectedIndexChanged(int index)
 void StageSelectorWidget::sessionStageChanged()
 {
     if (!_internalChange) {
-        auto index = _dropDown->findData(QVariant::fromValue(_sessionState->stage()));
+        size_t index = -1;
+        for (size_t i = 0; i < _dropDown->count(); i++) {
+            auto const& data = _dropDown->itemData(i);
+            auto        entry = data.value<SessionState::StageEntry>();
+            if (entry == _sessionState->stageEntry()) {
+                index = i;
+            }
+        }
         if (index != -1) {
             QSignalBlocker blocker(_dropDown);
             _dropDown->setCurrentIndex(index);
         }
     }
 }
 
-void StageSelectorWidget::stageRenamed(std::string const& name, PXR_NS::UsdStageRefPtr const& stage)
+void StageSelectorWidget::stageRenamed(
+    std::string const&              oldName,
+    SessionState::StageEntry const& renamedEntry)
 {
-    auto index = _dropDown->findData(QVariant::fromValue(stage));
+    size_t index = -1;
+    for (size_t i = 0; i < _dropDown->count(); i++) {
```

Comment:
QComboBox::count() return int and not size_t, so all of these would be wrong.  (in fact, size_t is unsigned, so comparision with -1 might build break on some platform)

Reply:
Good point! Nice catch!